drivers/snmp-ups.c

```diff
@@ -878,9 +940,10 @@ void su_setinfo(snmp_info_t *su_info_p, const char *value)
 	if (SU_TYPE(su_info_p) == SU_TYPE_CMD)
 		return;
 
-	/* ups.status and ups.alarm have special handling, not here! */
+	/* ups.status and {ups, Lx, outlet, outlet.group}.alarm have special
+	 * handling, not here! */
 	if ((strcasecmp(su_info_p->info_type, "ups.status"))
-		&& (strcasecmp(su_info_p->info_type, "ups.alarm")))
+		&& (strcasecmp(strrchr(su_info_p->info_type, '.'), ".alarm")))
```

Comment:
Do we have a guarantee (check somewhere) that "su_info_p->info_type" does not contain a "." other than expected separation of ".alarm", ".status" etc?